app/scripts/modules/azure/loadBalancer/configure/createLoadBalancer.controller.js

```diff
@@ -316,11 +310,33 @@ module.exports = angular.module('spinnaker.azure.loadBalancer.create.controller'
 
       $scope.taskMonitor.submit(
         function() {
-          let params = { cloudProvider: 'azure', providerType: 'azure', appName: application.name, clusterName: $scope.loadBalancer.clusterName,
+          let params = { cloudProvider: 'azure', appName: application.name, clusterName: $scope.loadBalancer.clusterName,
             resourceGroupName: $scope.loadBalancer.clusterName,
             loadBalancerName: $scope.loadBalancer.name
           };
-          return loadBalancerWriter.upsertLoadBalancer($scope.loadBalancer, application, descriptor, params);
+
+          var name = $scope.loadBalancer.clusterName || $scope.loadBalancer.name;
+          var probeName = name + '_probe';
+          var ruleNameBase = name + '_rule_';
+          $scope.loadBalancer.type = 'upsertLoadBalancer';
+          if (!$scope.loadBalancer.vnet && !$scope.loadBalancer.subnetType) {
+            $scope.loadBalancer.securityGroups = null;
+          }
+
+          $scope.loadBalancer.probes[0].probeName = probeName;
+
+          for(var i = 0; i < $scope.loadBalancer.loadBalancingRules.length; i++)
+          {
+            $scope.loadBalancer.loadBalancingRules[i].ruleName = ruleNameBase + i;
+            $scope.loadBalancer.loadBalancingRules[i].probeName = probeName;
+          }
```

Comment:
This can be rewritten using ES5's `forEach` method:
```javascript
$scope.loadBalancer.loadBalancingRules.forEach((rule, index) => {
  rule.ruleName = ruleNameBase + index;
  rule.probeName = probeName;
});
```
We are trying to use ES5/6 methods and syntax where we can.